docs/usage.rst

```diff
@@ -82,6 +82,64 @@ and then you would configure it in your settings:
 
     HAWK_CREDENTIALS_LOOKUP = 'yourapi.auth.hawk_lookup'
 
+Alternately, you can subclass ``HawkAuthentication`` and override the ``_credentials_lookup()`` method. For example:
+
+.. code-block:: python
+
+    class YourHawk(HawkAuthentication):
+        def _credentials_lookup(self, id):
+            user = HawkUser.objects.get(some_id=id)
+            return {
+                'id': id,
+                'key': user.key,
+                'algorithm': 'sha256'
+            }
+
+and then specify your new class instead in the authentication backend list:
+
+.. code-block:: python
+
+    REST_FRAMEWORK = {
+        'DEFAULT_AUTHENTICATION_CLASSES': (
+            'yourapi.auth.YourHawk',
+        ),
+        ...
+    }
+
+By default, a generic ``HawkAuthenticatedUser`` instance is returned when valid Hawk credentials are found. If you need another user model, you can set up a lookup function under the ``HAWK_USER_LOOKUP`` setting. This function receives the request and the matched credentials dict as parameters and returns a ``(user, auth)`` tuple as per `custom authentication`_. For example, with a ``HawkUser`` model whose ``user_id`` is included in the credentials dict, you can write a function ``hawk_user_lookup`` as follows:
+
+.. code-block:: python
+
+    def hawk_user_lookup(request, credentials):
+        return HawkUser.objects.get(some_id=credentials['user_id'])
+
+and then you would configure it in your settings:
+
+.. code-block:: python
+
+    HAWK_USER_LOOKUP = 'yourapi.auth.hawk_user_lookup'
+
+.. _`custom authentication`: http://www.django-rest-framework.org/api-guide/authentication/#custom-authentication
+
+Alternately, you can subclass ``HawkAuthentication`` and override the ``_user_lookup()`` method. For example:
+
+.. code-block:: python
+
+    class YourHawk(HawkAuthentication):
+        def _user_lookup(self, request, credentials):
+            return HawkUser.objects.get(some_id=credentials['user_id'])
+
+and then specify your new class instead in the authentication backend list:
```

Comment:
Hmm. I was originally splitting user/credential lookup because they're not necessarily coupled (say, fixed credentials to access a server, and a user id picked from the session or from a request header), and didn't want to imply that one _has_ to override both. I've read plenty of docs that combine things like that, and they usually give the impression that you have to change everything to change any one thing.